hibernate-core/src/main/java/org/hibernate/proxy/AbstractLazyInitializer.java

```diff
@@ -102,6 +104,19 @@ public final SharedSessionContractImplementor getSession() {
 		return session;
 	}
 
+	@Override
+	public final void setLockMode(LockMode lockMode) {
+		this.requestedLockMode = lockMode;
+	}
+
+	@Override
+	public final LockMode getLockMode() {
+		if ( requestedLockMode == null ) {
+			return LockMode.NONE;
+		}
+		return requestedLockMode;
```

Comment:
Shouldn't this only be called when uninitialized?

```suggestion
		assert isUninitialized();
		return requestedLockMode == null ? LockMode.NONE : requestedLockMode;
```